datastore.go

```diff
@@ -55,24 +68,23 @@ func NewDatastore(path string, opts *pebble.Options) (*Datastore, error) {
 // exists. If the key doesn't exist, ds.ErrNotFound will be returned. When no
 // error occurs, the size of the value is also returned.
 func (d *Datastore) get(key []byte, retval bool) ([]byte, int, error) {
-	val, closer, err := d.db.Get(key)
-	switch err {
-	case nil:
-		// do nothing
-	case pebble.ErrNotFound:
+	// We do not use db.Get because it bypasses bloom filters, resulting
+	// in more reads that needed, particularly for non existing keys.
+	// https://github.com/cockroachdb/pebble/issues/197
+	iter := d.db.NewIter(nil)
```

Comment:
Would it make sense to set `LowerBound` option here instead? That then removes the need for seek below.

Reply:
Does it make any difference at all to the performance of Seek? iirc it doesn't really do anything...

Reply:
A quick benchmark should offer some colour. From reading the code it seems slightly odd to not use `LowerBound` option considering its purpose is to limit the scope of iterator.

Reply:
I think the iterator iterates "automatically", until current >= LowerBound when doing Next(). I'm not even sure that it does a Seek() by itself. The user needs to do Seek() operations, and then you can use Bounds to fine tune from there.

In our case, since we are seeking directly to the value we are looking for, Bounds don´t matter.

Reply:
When doing it.First() or it.Next() https://github.com/cockroachdb/pebble/blob/fdf055ddb6e438a63b81ac2d751da8232b07c893/iterator.go#L1813 it will position the iterator to be within bounds, it does a normal SeekGE() in that case.

But as far as I can sees, this option is unused until then, and therefore does not matter for our use.